controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -360,53 +397,53 @@ func (r *StorageClassRequestReconciler) reconcilePhases() (reconcile.Result, err
 	return reconcile.Result{}, nil
 }
 
-func (r *StorageClassRequestReconciler) reconcileCephBlockPool(storageProfile *v1.StorageProfile) error {
-
-	failureDomain := r.storageCluster.Status.FailureDomain
+func (r *StorageClassRequestReconciler) reconcileCephBlockPool() error {
 
 	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephBlockPool, func() error {
-		if err := r.own(r.cephBlockPool); err != nil {
-			return err
-		}
-		deviceClass := storageProfile.Spec.DeviceClass
-		deviceSetList := r.storageCluster.Spec.StorageDeviceSets
-		var deviceSet *v1.StorageDeviceSet
-		for i := range deviceSetList {
-			ds := &deviceSetList[i]
-			// get the required deviceSetName of the profile
-			if deviceClass == ds.DeviceClass {
-				deviceSet = ds
-				break
-			}
-		}
+		r.cephBlockPool.Spec.PoolSpec.FailureDomain = r.storageCluster.Status.FailureDomain
+		return nil
+	})
 
-		if deviceSet == nil {
-			return fmt.Errorf("could not find device set with device class %q in storagecluster", deviceClass)
+	if err != nil {
+		r.log.Error(
+			err,
+			"Failed to update CephBlockPool.",
+			"CephBlockPool",
+			klog.KRef(r.cephBlockPool.Namespace, r.cephBlockPool.Name),
+		)
+		return err
+	}
+
+	phase := ""
+	if r.cephBlockPool.Status != nil {
+		phase = string(r.cephBlockPool.Status.Phase)
+	}
+
+	r.setCephResourceStatus(r.cephBlockPool.Name, "CephBlockPool", phase, nil)
+
+	return nil
+}
+
+func (r *StorageClassRequestReconciler) reconcileRadosNamespace() error {
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephRadosNamespace, func() error {
+		if err := r.own(r.cephRadosNamespace); err != nil {
+			return err
 		}
 
-		addLabel(r.cephBlockPool, controllers.StorageConsumerNameLabel, r.storageConsumer.Name)
-		addLabel(r.cephBlockPool, controllers.StorageProfileSpecLabel, storageProfile.GetSpecHash())
+		addLabel(r.cephRadosNamespace, controllers.StorageConsumerNameLabel, r.storageConsumer.Name)
 
-		r.cephBlockPool.Spec = rookCephv1.NamedBlockPoolSpec{
-			PoolSpec: rookCephv1.PoolSpec{
-				FailureDomain: failureDomain,
-				DeviceClass:   deviceClass,
-				Replicated: rookCephv1.ReplicatedSpec{
-					Size:                     3,
-					ReplicasPerFailureDomain: 1,
-				},
-				Parameters: storageProfile.Spec.BlockPoolConfiguration.Parameters,
-			},
+		r.cephRadosNamespace.Spec = rookCephv1.CephBlockPoolRadosNamespaceSpec{
+			BlockPoolName: r.cephBlockPool.Name,
```

Comment:
It will be better to read the block pool name from the source of truth, not from the block pool. 
The source of truth is the `r.StorageClassRequest.Spec.StorageProfile` or if empty, the default blook pool, which in ODF has a fixed and  known name (I belive defined as a constant in some other file)

Reply:
The default block pool does not have a constant name, it's generated based on the StorageCluster name. And as I've already said, per my understanding of not only the existing provider mode behavior but the new design, there shouldn't be a default block pool existing at all, nor am I being allowed to create it if it doesn't exist.